src/Confirmation.vala

```diff
@@ -19,18 +19,51 @@
 */
 
 public class Notifications.Confirmation : AbstractBubble {
+    private string _confirmation_type;
+    public string confirmation_type {
+        get {
+            return _confirmation_type;
+        }
+        set {
+            if (value == _confirmation_type) {
+                return;
+            }
+
+            _confirmation_type = value;
+
+            if (icon_name_binding != null) {
+                icon_name_binding.unbind ();
+            }
+            if (progress_binding != null) {
+                progress_binding.unbind ();
+            }
```

Comment:
you don't need to keep track of the binding if you destroy the old contents.

Reply:
But we can't just destroy the old contents? The old contents should slide down, and if it's destroyed there will be no animation for this.